test/BreakEventHandler.js

```diff
@@ -0,0 +1,93 @@
+var expect = require('chai').expect,
+    launcher = require('./helpers/launcher'),
+    InjectorClient = require('../lib/InjectorClient').InjectorClient,
+    BreakEventHandler = require('../lib/BreakEventHandler').BreakEventHandler,
+    ScriptManager = require('../lib/ScriptManager').ScriptManager;
+    
+describe('BreakEventHandler', function() {
+  var childProcess,
+      session;
+
+  describe('should cache events', function() {
+    it('if started in debug-brk mode', function(done) {
+      prepareBreakEventHandler(true, function() {
+        expect(session.breakEventHandler._eventsCache).to.instanceof(Array);
+        expect(session.breakEventHandler._eventsCache.length).to.equal(0);
+        
+        session.debuggerClient.request('continue', undefined, function() {
+          childProcess.stdin.write('pause\n');
+          setTimeout(function() {
+            expect(session.breakEventHandler._eventsCache.length).to.equal(1);
+            done();
+          }, 100);
+        });
+      });
+    });
+    
+    it('if started in debug mode', function(done) {
+      prepareBreakEventHandler(false, function() {
+        expect(session.breakEventHandler._eventsCache).to.instanceof(Array);
+        expect(session.breakEventHandler._eventsCache.length).to.equal(0);
+        
+        childProcess.stdin.write('pause\n');
+        setTimeout(function() {
+          expect(session.breakEventHandler._eventsCache.length).to.equal(1);
+          done();
+        }, 100);
+      });
+    });
+  });
+  
+  describe('should resolve events', function() {
+    before(function(done) {
+      prepareBreakEventHandler(false, function() {
+        childProcess.stdin.write('pause\n');
+        setTimeout(done, 100);
+      });
+    });
+    
+    it('on resource-tree-resolved', function(done) {
+      session.frontendClient.once('Debugger.paused', function() {
+        done();
+      });
+      
+      session.debuggerClient.request('scripts', null, function(error, result) {
+        if (error) done(error);
+
+        result.forEach(session.scriptManager.addScript.bind(session.scriptManager));
+        session.resourceTreeResolved = true;
+        session.emit('resource-tree-resolved');
+      });
+    });
+    
+    it('immediately after resource-tree-resolved', function(done) {
+      session.frontendClient.once('Debugger.paused', function() {
+        done();
+      });
+
+      session.debuggerClient.request('continue', null, function() {
+        childProcess.stdin.write('pause\n');
+      });
+    });
+  });
```

Comment:
IIUC, this test is relying on the global state changes made by the previous test and checking that continue + pause will be correctly handled. It fails on timeout when the implementation is not correct. If that's the case, then again add an explanatory comment please.